### ISSUE TITLE
editable column setting changed

### ISSUE THREAD

**Issue by uap-universe** [dj-fiorex/angular2-smart-table#30]
Consider this code:
~~~Typescript
this.isSortable = typeof this.settings['sort'] === 'undefined' ? true : !!this.settings['sort'];
this.isEditable = typeof this.settings['isEditable'] === 'undefined' ? true : !!this.settings['isEditable'];
~~~

We should either
* rename both settings and include that change in the migration guide (#29 )
* revert the change for editable (the original name was `editable`)

**Comment by uap-universe** [comment]
@dj-fiorex I did not look into the other settings, yet. Maybe you can give me some info from your memory what may have also changed and decide which changes persist and which should be reverted for compatibility. Then I could include this in the 2.1.0 milestone.

**Comment by dj-fiorex** [comment]
I think we need to write a compatibility line beetween this new fork and the old one. I think that the original team doesn't want to work on the old project anymore so we are free to make all the changes we deem appropriate to improve this project.
So for this question specifically, if i remember correctly there are some more settings key that need to be fixed but it's like 4 months that i do not read this code

**Comment by uap-universe** [comment]
Agreed. Unfortunately I have a release deadline for my project and I am quite busy at the moment. We are not in a hurry, though, so I think we can take our time to sort these things out properly.

I would vote for `editable`, `sortable` etc. as setting key names. The reason is, that `isEditable` looks like a function. If it *was* a function, the name would be fine. And maybe we add this is a feature later. Then we would have `editable: boolean` as simple setting to statically make a column editable and we would also have `isEditable: (??)->boolean` with some function for which I am not yet sure which parameter it should have. But maybe it's nonsense, I haven't thought about that carefully enough, yet.

In my project I have a similar use case for `hide`. In the original implementation there was nothing like `hide` as far as I remember. So we just deleted the columns from the settings when we wanted to hide a column. In the future we could imagine two settings: `hide` (or `hidden` to keep the style of `editable`, `sortable`, `filterable` etc.) for static hide/show and `isHidden` as a function to dynamically hide a column depending on some properties. But again, I haven't thought enough about that and maybe it would be too much of a performance impact to dynamically recall these functions to update the settings.

**Comment by uap-universe** [comment]
I added compatibility for the deprecated setting keys `editable` and `sort`. If we find that there are more keys that should have a compatibility layer, we can easily add them.

For the time being I dropped the other wild ideas from my previous post.
